test/Microsoft.ML.Tests/MemoryCollectionTests.cs

```diff
@@ -0,0 +1,176 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.ML;
+using Microsoft.ML.Runtime;
+using Microsoft.ML.Runtime.Api;
+using Microsoft.ML.Runtime.Data;
+using Microsoft.ML.TestFramework;
+using Microsoft.ML.Trainers;
+using Microsoft.ML.Transforms;
+using System;
+using System.Collections.Generic;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Microsoft.ML.EntryPoints.Tests
+{
+    public class MemoryCollectionTests : BaseTestClass
+    {
+        public MemoryCollectionTests(ITestOutputHelper output)
+            : base(output)
+        {
+
+        }
+
+        [Fact]
+        public void ConstructorDoesntThrow()
+        {
+            Assert.NotNull(new MemoryCollection<Input>(new List<Input>()));
+            Assert.NotNull(new MemoryCollection<Input>(new List<Input>() { new Input { Number1 = 1, String1 = "1" } }));
+            Assert.NotNull(new MemoryCollection<Input>(new Input[0]));
+            Assert.NotNull(new MemoryCollection<Input>(new Input[1] { new Input { Number1 = 1, String1 = "1" } }));
+            Assert.NotNull(new MemoryCollection<Input>(null));
+        }
+
+        [Fact]
+        public void CanSuccessfullyApplyATransform()
+        {
+            var collection = new MemoryCollection<Input>(new List<Input>() { new Input { Number1 = 1, String1 = "1" } });
+
+            using (var environment = new TlcEnvironment())
+            {
+                Experiment experiment = environment.CreateExperiment();
+                ILearningPipelineDataStep output = collection.ApplyStep(null, experiment) as ILearningPipelineDataStep;
+
+                Assert.NotNull(output.Data);
```

Comment:
>Assert.NotNull(output.Data); [](start = 16, length = 28)

If you really meant the `as` above, you should first `Assert.NonNull` on `output`. If you *didn't* mean the `as` above, then you should do a direct `()` style cast to `ILearningPipelineDataStep`. The only reason to use an `as` style cast if you are entertaining the possibility that it *might* not implement that interface, in which case, given that this is a test, you should test that.